### PR TITLE
Support fetching data with Request object

### PR DESCRIPTION
**Pull request by ShawnMa16**

### CODE REVIEW

File: Sources/FioriIntegrationCards/Components/Common/Model/Manifest.swift [hunk 1]
```diff
@@ -16,6 +17,8 @@ public class Manifest: Decodable, Identifiable, ObservableObject {
         return app.id
     }
     
+    public var baseURL: String?
```
Comment (MarcoEidinger) [inline Sources/FioriIntegrationCards/Components/Common/Model/Manifest.swift:20]: `baseURL` is of type String?

File: Sources/FioriIntegrationCards/Components/Common/Model/DataFetcher.swift [hunk 2]
```diff
@@ -40,8 +40,8 @@ public class DataFetcher: Decodable {
         load()
     }
     
-    public func load() {
-        request?.send()
+    public func load(baseURL: String? = nil) {
```
Comment (MarcoEidinger) [inline Sources/FioriIntegrationCards/Components/Common/Model/DataFetcher.swift:43]: `baseURL` is of type String?

File: Sources/FioriIntegrationCards/Components/Common/Model/Card.swift [hunk 3]
```diff
@@ -65,6 +65,21 @@ public enum Card: Decodable, Identifiable, CustomStringConvertible {
             throw DecodingError.dataCorrupted(context)
         }
     }
+
+    public func loadDataIfNeeded(baseURL: String?) {
```
Comment (MarcoEidinger) [inline Sources/FioriIntegrationCards/Components/Common/Model/Card.swift:69]: `baseURL` is of type String?

File: Sources/FioriIntegrationCards/Components/Common/Model/BaseCard.swift [hunk 4]
```diff
@@ -138,7 +157,7 @@ open class BaseBaseCard: Decodable, ObservableObject, Identifiable {
     
     public let headerPublisher = CurrentValueSubject<CurrentValueSubject<(Data, String?)?, Never>?, Never>(nil)
     public let contentPublisher = CurrentValueSubject<CurrentValueSubject<(Data, String?)?, Never>?, Never>(nil)
-    
+    public let baseURL: CurrentValueSubject<String?, Never> = CurrentValueSubject<String?, Never>(nil)
```
Comment (MarcoEidinger) [inline Sources/FioriIntegrationCards/Components/Common/Model/BaseCard.swift:160]: `baseURL` is of type String?

File: Sources/FioriIntegrationCards/Components/Common/Model/Manifest.swift [hunk 5]
```diff
@@ -29,6 +32,50 @@ public class Manifest: Decodable, Identifiable, ObservableObject {
...
+        let _path = path.appendingPathComponent("manifest.json")
+        do {
+            let data = try Data(contentsOf: _path)
+            _model = try JSONDecoder().decode(Manifest.self, from: data)
+        } catch {
+            print(error)
+        }
+        self.app        = _model.app
+        self.card       = _model.card
+        self.baseURL    = path.absoluteString
```
Comment (MarcoEidinger) [inline Sources/FioriIntegrationCards/Components/Common/Model/Manifest.swift:47]: converting URL to String

File: Sources/FioriIntegrationCards/Components/Common/Model/Request.swift [hunk 6]
```diff
@@ -43,20 +43,60 @@ class Request: Decodable {
...
+            print(error)
+            return nil
+        }
+    }
+    
+    func loadDataFromNetwork(baseURL: String?) {
+        var urlRequest: URLRequest!
+        do {
+            if let _baseURL = baseURL {
+                urlRequest = try NetworkRouter.getURLRequest(with: self, baseURL: URL(string: _baseURL))
```
Comment (MarcoEidinger) [inline Sources/FioriIntegrationCards/Components/Common/Model/Request.swift:83]: finally we load data with `baseURL` of type URL

File: Sources/FioriIntegrationCards/Components/Common/Model/Request.swift [hunk 7]
```diff
@@ -43,20 +43,60 @@ class Request: Decodable {
...
+                        self.fetchedData.send(data)
+                    } else {
+                        loadDataFromNetwork(baseURL: baseURL)
+                    }
+                }
+            }
+        }
+    }
+    
+    func getDataFromBundle(bundleURL: URL) -> Data? {
```
Comment (MarcoEidinger) [inline Sources/FioriIntegrationCards/Components/Common/Model/Request.swift:66]: finally we load data with `bundleURL` of type URL

File: Sources/FioriIntegrationCards/Utils/String+Extensions.swift [hunk 8]
```diff
@@ -9,6 +9,16 @@ import Foundation
 
 extension String {
     
+    var isValidURL: Bool {
+        let detector = try! NSDataDetector(types: NSTextCheckingResult.CheckingType.link.rawValue)
```
Comment (MarcoEidinger) [inline Sources/FioriIntegrationCards/Utils/String+Extensions.swift:13]: TODO: better guard condition?

**Comment by MarcoEidinger** [comment]
@ShawnMa16 please remove the respective limitation note in README.md as your PR resolves it :)

> FioriIntegrationCards currently handles only data which is in-line json; must be augmented to support resolving relative data files, and remote URIs

**Comment by ShawnMa16** [comment]
> @ShawnMa16 please remove the respective limitation note in README.md as your PR resolves it :)
> 
> > FioriIntegrationCards currently handles only data which is in-line json; must be augmented to support resolving relative data files, and remote URIs

Hey Marco, I've addressed the comments, can you take a look?

**Comment by MarcoEidinger** [comment]
Thanks @ShawnMa16 ,  looks good. This PR is good for merging. I noticed that in some cases errors are getting logged via `print` and are not propagated to the consumer. As this is already the case in other code for integration cards I created https://github.com/SAP/cloud-sdk-ios-fiori/issues/71 to address this in one shot.

**Comment by MarcoEidinger** [comment]
Closes #33

**Comment by MarcoEidinger** [comment]
I squashed commits to one and kept the best commit description :)

**Pull request merged**